armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -182,6 +199,145 @@ def _isTopDummyBlockPresent(self):
+                            # so just set the bottom of the component to
+                            # the top of the target component in the block below it
+                            targetCompForBlockBelow = (
+                                self.expansionData.getTargetComponent(
+                                    self.linked.linkedBlocks[b][0]
+                                )
+                            )
+                            c.zbottom = targetCompForBlockBelow.ztop
+                    c.ztop = c.zbottom + c.height
+
+        # align downward expanding components
+        # new c.height is set, but c.zbottom and c.ztop need to be set
+        blockList = list(componentsToExpandDown.keys())
+        componentPerBlockList = list(componentsToExpandDown.values())
+        for ib, b in reversed(list(enumerate(blockList))):
+            for c in componentPerBlockList[ib]:
+                if ib == len(blockList) - 1:
+                    c.ztop = topAnchorForDownwardExpansion
+                else:
+                    if self.linked.linkedComponents[c][1] is not None:
+                        # use linked components above
+                        c.ztop = self.linked.linkedComponents[c][1].zbottom
+                    else:
+                        # otherwise there aren't any linked components
+                        # so just set the top of the component to
+                        # the bottom of target component in the block above it
+                        targetCompForBlockAbove = self.expansionData.getTargetComponent(
+                            self.linked.linkedBlocks[b][1]
+                        )
+                        c.ztop = targetCompForBlockAbove.zbottom
+                c.zbottom = c.ztop - c.height
+
+        # align upward expanding pin components within control rod bundle.
+        # new c.height has been updated, but c.zbottom and c.ztop need to be set
+        for ib, b in enumerate(self.linked.a.getChildrenWithFlags(Flags.CONTROL)):
+            c = _getComponent(b, Flags.CONTROL)
+            if ib == 0:
+                cladComp = _getComponent(b, Flags.CLAD)
+                c.zbottom = cladComp.zbottom
+            else:
+                c.zbottom = self.linked.linkedComponents[c][0].ztop
+            c.ztop = c.zbottom + c.height
+
+        # redistribute block bounds based on the target component
+        adjustedLowestControlDuct = False
+        for ib, b in enumerate(self.linked.a):
+            c = self.expansionData.getTargetComponent(b)
+            runLog.debug("      Component {0} is target component".format(c))
+            if 0 < ib < numOfBlocks - 1:
+                b.p.zbottom = c.zbottom
+                # correct potentially misaligned blocks
+                if (
+                    self.linked.linkedBlocks[b][0].p.ztop != c.zbottom
+                    and ib < numOfBlocks - 1
+                ):
+                    b.p.zbottom = self.linked.linkedBlocks[b][0].p.ztop
```

Comment:
trying to wrap my head around lines 308-312:

If the target component's `c.zbottom` does not align with `b.p.ztop` of the block below, then we updated `b.p.zbottom` to match, but not `c.zbottom`.

I'm having a hard time figuring out whether we want `b.p.zbottom` and `c.zbottom` to be out-of-sync here. I know `c.zbottom` just gets trashed after this function, but it could potentially be used in line 317 below.

Reply:
Okay, I have a clearer understanding of what's happening here now. Since we're just iterating through once and `cTmp.zbottom` is from the component above the current block, then it doesn't matter that `c.zbottom` is not updated with `b.p.zbottom` in line 312, because it's not going to be used again after this.